### PR TITLE
Self managed nodes move oke from exp

### LINKED ISSUE #289: Add support for OKE self managed nodes
**What would you like to be added**:

Add support for OKE self managed nodes - https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengworkingwithselfmanagednodes.htm

We can do this via 2 mechanism 

1. Single node
2. Machinepool via Instance Pool

**Why is this needed**:

This will help OKE customers use advantages of CAPI to automate provisioning/updating of self managed nodes.

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
OKE self managed nodes and remove experimental of cluster and control plane

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #289

### CODE REVIEW

File: test/e2e/managed_cluster_test.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ package e2e
 import (
 	"context"
 	"fmt"
+	infrastructurev1beta2 "github.com/oracle/cluster-api-provider-oci/api/v1beta2"
```
Comment (joekr) [inline test/e2e/managed_cluster_test.go:25]: Should this move down out of the built in library group?

File: cloud/scope/managed_control_plane.go [hunk 2]
```diff
@@ -46,6 +47,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	OKEInitScript = "#!/usr/bin/env bash\nbash /etc/oke/oke-install.sh \\\n  --apiserver-endpoint \"CLUSTER_ENDPOINT\" \\\n  --kubelet-ca-cert \"BASE_64_CA\""
```
Comment (joekr) [inline cloud/scope/managed_control_plane.go:51]: What about calling out https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengcloudinitforselfmanagednodes.htm as the guide for where this script came from?

File: cloud/scope/vnic_reconciler.go [hunk 3]
```diff
@@ -65,7 +65,7 @@ func (m *MachineScope) createVnicAttachment(ctx context.Context, spec infrastruc
 		}
 	}
 
-	tags := m.getFreeFormTags(*m.OCICluster)
+	tags := m.getFreeFormTags()
```
Comment (joekr) [inline cloud/scope/vnic_reconciler.go:68]: I may have missed this why isn't the cluster being passed?

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ocimanagedclusters.yaml [hunk 4]
```diff
@@ -108,6 +108,7 @@ spec:
                     description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
                     type: string
                 type: object
+                x-kubernetes-map-type: atomic
```
Comment (joekr) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_ocimanagedclusters.yaml:111]: more for my own learning: What is setting these new server-side apply fields?

File: docs/src/managed/self-managed-nodes.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
+# Self managed nodes
+CAPOCI supports OKE self managed nodes. With this feature, CAPI features such as [rolling update][capi-upgrade],
```
Comment (joekr) [inline docs/src/managed/self-managed-nodes.md:2]:
```suggestion
CAPOCI supports [OKE self managed nodes][self-managed-nodes]. With this feature, CAPI features such as [rolling update][capi-upgrade],
```

File: exp/api/v1beta2/ocimanagedmachinepool_webhook.go [hunk 6]
```diff
@@ -18,6 +18,7 @@ package v1beta2
 
 import (
 	"fmt"
+	infrastructurev1beta2 "github.com/oracle/cluster-api-provider-oci/api/v1beta2"
```
Comment (joekr) [inline exp/api/v1beta2/ocimanagedmachinepool_webhook.go:21]: move the import down.

File: exp/api/v1beta2/ocimanagedmachinepool_webhook_test.go [hunk 7]
```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package v1beta2
 
 import (
+	infrastructurev1beta2 "github.com/oracle/cluster-api-provider-oci/api/v1beta2"
```
Comment (joekr) [inline exp/api/v1beta2/ocimanagedmachinepool_webhook_test.go:20]: move import down

File: exp/controllers/ocimachinepool_controller.go [hunk 8]
```diff
@@ -19,6 +19,7 @@ package controllers
 import (
 	"context"
 	"fmt"
+	expV1Beta1 "github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1"
```
Comment (joekr) [inline exp/controllers/ocimachinepool_controller.go:22]: move import down

File: exp/controllers/ocivirtual_machinepool_controller_test.go [hunk 9]
```diff
@@ -18,6 +18,7 @@ package controllers
 
 import (
 	"context"
+	infrastructurev1beta2 "github.com/oracle/cluster-api-provider-oci/api/v1beta2"
```
Comment (joekr) [inline exp/controllers/ocivirtual_machinepool_controller_test.go:21]: move import down

**Review by joekr — CHANGES_REQUESTED** [review]
very small nits. This is awesome

**Comment by shyamradhakrishnan** [comment]
unit test
```
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	63.357s	coverage: 23.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.064s	coverage: 15.7% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/metrics	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.007s	coverage: 19.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	241.856s	coverage: 75.1% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.091s	coverage: 76.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.803s	coverage: 60.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	15.682s	coverage: 16.5% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.020s	coverage: 5.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	0.613s	coverage: 59.1% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
go tool cover -func=coverage.out -o coverage.txt
go tool cover -html=coverage.out -o coverage.html
```

File: cloud/scope/vnic_reconciler.go (hunk 3 above)
Comment (shyamradhakrishnan) [reply cloud/scope/vnic_reconciler.go:68]: we changed impl in machine.go getFreeFormTags method to use cluster accessor

File: docs/src/managed/self-managed-nodes.md (hunk 5 above)
Comment (shyamradhakrishnan) [reply docs/src/managed/self-managed-nodes.md]: fixed

File: exp/api/v1beta2/ocimanagedmachinepool_webhook.go (hunk 6 above)
Comment (shyamradhakrishnan) [reply exp/api/v1beta2/ocimanagedmachinepool_webhook.go]: fixed

File: exp/api/v1beta2/ocimanagedmachinepool_webhook_test.go (hunk 7 above)
Comment (shyamradhakrishnan) [reply exp/api/v1beta2/ocimanagedmachinepool_webhook_test.go]: fixed

File: exp/controllers/ocimachinepool_controller.go (hunk 8 above)
Comment (shyamradhakrishnan) [reply exp/controllers/ocimachinepool_controller.go]: fixed

File: exp/controllers/ocivirtual_machinepool_controller_test.go (hunk 9 above)
Comment (shyamradhakrishnan) [reply exp/controllers/ocivirtual_machinepool_controller_test.go]: fixed

File: config/crd/bases/infrastructure.cluster.x-k8s.io_ocimanagedclusters.yaml (hunk 4 above)
Comment (shyamradhakrishnan) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_ocimanagedclusters.yaml:111]: not sure why it did not add earlier and added now when we moved the spec, it is present in OCICluster identityRef earlier itself https://github.com/oracle/cluster-api-provider-oci/blob/main/config/crd/bases/infrastructure.cluster.x-k8s.io_ociclusters.yaml#L108

**Comment by shyamradhakrishnan** [comment]
e2e 
```
  Timeline >>
  STEP: Tearing down the management cluster @ 07/06/23 05:43:33.548
  INFO: Error getting pod capi-kubeadm-bootstrap-system/capi-kubeadm-bootstrap-controller-manager-5996c56969-gbsrp, container manager: Get "https://127.0.0.1:40795/api/v1/namespaces/capi-kubeadm-bootstrap-system/pods/capi-kubeadm-bootstrap-controller-manager-5996c56969-gbsrp": dial tcp 127.0.0.1:40795: connect: connection refused
  INFO: Error getting pod capi-kubeadm-control-plane-system/capi-kubeadm-control-plane-controller-manager-85f84fcdcb-jfvn9, container manager: Get "https://127.0.0.1:40795/api/v1/namespaces/capi-kubeadm-control-plane-system/pods/capi-kubeadm-control-plane-controller-manager-85f84fcdcb-jfvn9": dial tcp 127.0.0.1:40795: connect: connection refused
  INFO: Error getting pod capi-system/capi-controller-manager-5fbc569749-g8fbl, container manager: Get "https://127.0.0.1:40795/api/v1/namespaces/capi-system/pods/capi-controller-manager-5fbc569749-g8fbl": dial tcp 127.0.0.1:40795: connect: connection refused
  INFO: Error getting pod cluster-api-provider-oci-system/capoci-controller-manager-7ddbd9f6bb-8c7w5, container manager: Get "https://127.0.0.1:40795/api/v1/namespaces/cluster-api-provider-oci-system/pods/capoci-controller-manager-7ddbd9f6bb-8c7w5": dial tcp 127.0.0.1:40795: connect: connection refused
  << Timeline
------------------------------

Ran 7 of 28 Specs in 3263.654 seconds
SUCCESS! -- 7 Passed | 0 Failed | 0 Pending | 21 Skipped
```

**Comment by shyamradhakrishnan** [comment]
self managed e2e test
```
Managed Workload cluster creation Managed Cluster - Self managed nodes
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/managed_cluster_test.go:325
Ran 1 of 28 Specs in 1344.825 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 27 Skipped
```

**Review by joekr — APPROVED** [review]

**Pull request merged**